lib/src/barback/asset_environment.dart

```diff
@@ -445,27 +482,30 @@ class AssetEnvironment {
     return directories.any((dir) => path.isWithin(dir, sourcePath));
   }
 
+  /// Whether or not are currently pausing updates, based on a call to
+  /// [pauseUpdates].
+  ///
+  /// Can be resumed with a call to [resumeUpdates].
+  bool _updatesPaused = false;
+
   /// Pauses sending source asset updates to barback.
   void pauseUpdates() {
-    // Cannot pause while already paused.
-    assert(_modifiedSources == null);
+    assert(_updatesPaused == false);
```

Comment:
[nit] `assert(!_updatesPaused)`

`== false` and `== true` feels weird outside of loose types.